### PR TITLE
FeatureFlags: Use interface rather than manager

### PR DESCRIPTION
**Pull request by ryantxu**
**What is this feature?**

This PR changes most things to use `featuremgmt.FeatureToggles` rather than `*featuremgmt.FeatureManager`.  The manager should only be used when we need more detailed control/info from the toggles.

This also unifies the `plugins.FeatureToggles` with `featuremgmt.FeatureToggles`, so we only have one interface.

**Why do we need this feature?**

As the manager work expands, we do not want that to touch everything.

**Who is this feature for?**

internal developers

### CODE REVIEW

File: pkg/plugins/ifaces.go [hunk 1]
```diff
@@ -149,11 +149,6 @@ func (fn ClientMiddlewareFunc) CreateClientMiddleware(next Client) Client {
 	return fn(next)
 }
 
-type FeatureToggles interface {
```
Comment (ryantxu) [inline pkg/plugins/ifaces.go:152]: @wbrowne -- we can now use `featuremgmt.FeatureToggles` directly

File: pkg/api/http_server.go [hunk 2]
```diff
@@ -126,7 +126,8 @@ type HTTPServer struct {
 	RouteRegister                routing.RouteRegister
 	RenderService                rendering.Service
 	Cfg                          *setting.Cfg
-	Features                     *featuremgmt.FeatureManager
+	Features                     featuremgmt.FeatureToggles
+	featureManager               *featuremgmt.FeatureManager
```
Comment (ryantxu) [inline pkg/api/http_server.go:130]: ^^ this is the most awkward change.  Really points out that this file is the messy jumble of everything 🤮

**Comment by marefr** [comment]
While you're at it maybe a good opportunity to split interfaces and implementations in separate packages like https://github.com/grafana/grafana/blob/main/contribute/backend/package-hierarchy.md#practical-example suggests and a package structure that other services uses? This might allow plugins and other things to reuse the interface without depending on the implementation/other dependencies the implementation imports.

File: pkg/api/http_server.go [hunk 3]
```diff
@@ -291,6 +292,7 @@ func ProvideHTTPServer(opts ServerOptions, cfg *setting.Cfg, routeRegister routi
 		QueryHistoryService:          queryHistoryService,
 		CorrelationsService:          correlationsService,
 		Features:                     features,
+		featureManager:               features, // the manager implements both, but two variables encourages
```
Comment (kylebrandt) [inline pkg/api/http_server.go:295]: Incomplete comment?

Comment (ryantxu) [reply pkg/api/http_server.go:295]: dooh 🤦🏻

**Comment by ryantxu** [comment]
> maybe a good opportunity to split interfaces and implementations in separate packages like

@marefr -- will address this some follow up PRs -- in particular, trying to split out the startup vs runtime (per/tenant) configuration and how apps/plugins can use them

**Review by zserge — APPROVED** [review]

**Review by marefr — APPROVED** [review]
LGTM

**Pull request merged**